src/test/java/com/atlassian/db/replica/it/DualConnectionIT.java

```diff
@@ -79,6 +79,22 @@ public void shouldRunNextValOnMainDatabase() throws SQLException {
         }
     }
 
+    @Test
+    public void shouldChangeTransactionIsolationLevel() throws SQLException {
+        try (PostgresConnectionProvider connectionProvider = new PostgresConnectionProvider()) {
+            final Connection connection = DualConnection.builder(
+                connectionProvider,
+                CircularConsistency.permanentConsistency().build()
+            ).build();
+
+            connection.setTransactionIsolation(Connection.TRANSACTION_READ_UNCOMMITTED);
+            connection.setAutoCommit(false);
```

Comment:
What if autoCommit=true? Will it still work fine?

Reply:
Yes. It just reproduces exactly the issue we seen on the production. There's a more generic unit test that targets the root cause.